Packages/com.vrchat.ClientSim/Runtime/CameraStacking/ClientSimStackedVRCameraSystem.cs

```diff
@@ -0,0 +1,149 @@
+using VRC.Core;
+using VRC.SDK3.ClientSim;
+
+namespace VRC.SDK3.ClientSim
+{
+    public class ClientSimStackedVRCameraSystem : MonoBehaviour
+    {
+        public ClientSimStackedCamera[] CameraStack;
+        [HideInInspector] public bool IsCameraStackingEnabled = false;
+        
+        private Camera MainSceneCamera;
+        private bool IsInitialized = false;
+        private bool IsReady = false;
+        private List<Camera> Cameras;
+        
+        private ClientSimMenu clientSimMenu;
+
+        public void Initialize(Camera playerCamera, ClientSimMenu menu)
+        {
+            MainSceneCamera = playerCamera;
+            clientSimMenu = menu;
+        }
+
+        public void Ready()
+        {
+            IsReady = true;
+        }
+        
+        void Update()
+        {
+            if(!IsReady) return;
+            if (!IsInitialized) { InitializeStackedSystem(); }
+            else
+            {
+                if (!Cameras[0].enabled)
+                {
+                    Debug.LogError("Stacked Cameras are not enabled.");
+                }
+            }
+        }
+     
+        /*void OnEnable()
+        {
+            #if !UNITY_ANDROID
+            if (MainSceneCamera != null)
+            {
+                if (!IsCameraStackingEnabled)
+                    CreateCameraStack();
+            }
+            #endif
+        }
+*/
+        void OnDisable()
+        {
+            #if !UNITY_ANDROID
+            if (MainSceneCamera != null)
+            {
+                if (IsCameraStackingEnabled)
+                    DestroyCameraStack();
+            }
+            #endif
+        }
+
+        public void InitializeStackedSystem()
+        {
+            #if !UNITY_ANDROID
+            Cameras = new List<Camera>();
+            if (MainSceneCamera != null)
+            {
+                CreateCameraStack();
+                IsInitialized = true;
+                FormattedLog("Camera Stack has been initialized.");
+            }
+            #else
+            gameObject.SetActive(false);
+            #endif
+        }
+
+        private void CreateCameraStack()
+        {
+            for (int i = 0; i < CameraStack.Length; i++)
+            {
+                AddCamera(i);
+            }
+            IsCameraStackingEnabled = true;
+        }
+
+        private void DestroyCameraStack()
+        {
+            for (int i = 0; i < Cameras.Count; i++)
+            {
+                DestroyCamera(i);
+            }
+            IsCameraStackingEnabled = false;
+        }
+
+        private void AddCamera(int index)
+        {
+            GameObject cameraObj = Instantiate(new GameObject(), MainSceneCamera.transform);
+            Camera cam = cameraObj.AddComponent<Camera>();
+            XRDevice.DisableAutoXRCameraTracking(cam, true);
+
+            cam.CopyFrom(MainSceneCamera); // Start by copying all the settings from the main camera
+            #if VRC_VR_STEAM // We only want this on SteamVR.
```

Comment:
Maybe I'm missing something, but I don't understand why we're handling XRDevice and VRC_VR_STEAM stuff here. I thought this would all only run in the editor, in 2D mode?

Reply:
this code is based on the client camera stacking, and so leaving in this VR stuff seemed appropriate, like you said for future in-Editor VR support